plugins/ltac2/g_ltac2.mlg

```diff
@@ -717,6 +717,9 @@ GRAMMAR EXTEND Gram
       | -> { CAst.make ~loc None }
     ]]
   ;
+  q_orient:
+    [ [ ori = ltac2_orient -> { ori } ] ]
+  ;
```

Comment:
Is it necessary to introduce `q_orient` in addition to `ltac2_orient`?  Seems like one nonterminal would be sufficient.

Reply:
Initially, I went this route: I renamed `ltac2_orient` with `q_orient`. But then the `docgram` tool gave me the following grammar (pseudo-code), due to the `OPTINREF` global rule:

```
q_orient :: <- | ->

rewriting :: OPT q_orient [...]
```

The problem is this is incorrect, as `q_orient` itself allows for an empty production (i.e. `OPT` should be in `q_orient`). Adding a redundant symbol fixed it, giving:

```
q_orient :: OPT (<- | ->)

rewriting :: q_orient [...]
```

(note that symbol `ltac2_orient` is spliced)

Reply:
Your initial plan was the right one.  For ease of use, we want the reader to see that q_orient is optional without having to follow the hyperlink.  We intentionally do this throughout `orderedGrammar`, which is meant to be equivalent to `fullGrammar`, but presented in a user-friendly way.

Reply:
Fair enough.  There's no reason there can't be an OPT in both q_orient and ltac2_oriented_rewriter.  To do this, use the initial form and add the following after the OPTINREF in `common.edit_mlg` to re-add the empty production:

```
q_orient: [
|
]
```

Though this is a bit nit-picky.

Reply:
But maybe it should be this edit instead:

```
q_orient: [
| REPLACE "<-" | "->"
| WITH OPT ("<-" | "->")
]
```

Reply:
Should be good now with the latest push... Note that I used the following as grammar edit:

```
q_orient: [
| DELETE "<-"
| REPLACE "->"
| WITH OPT ["->" | "<-"]
]
``` 

because I guess you cannot replace two productions in one go. Let me know if not idiomatic

Reply:
Correct.  This is fine.